### PR TITLE
fix(ivy): add support for optional nullable injection tokens

### PR DESCRIPTION
**Pull request by IgorMinar**
FW-778 #resolve

### CODE REVIEW

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/src/ngtsc/metadata/test/reflector_spec.ts [hunk 1]
```diff
@@ -185,6 +213,10 @@ function expectParameter(
 }
 
 function argExpressionToString(name: ts.Node): string {
+  if (name == null) {
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/compiler-cli/src/ngtsc/metadata/test/reflector_spec.ts]: done

**Comment by IgorMinar** [comment]
http://test/OCL:224598137:BASE:224598375:1544229687304:f6156df6

File: packages/compiler-cli/src/ngtsc/metadata/test/reflector_spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfaproject) [reply packages/compiler-cli/src/ngtsc/metadata/test/reflector_spec.ts]: When does it return `null`?

[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/compiler-cli/src/ngtsc/metadata/test/reflector_spec.ts]: this is just a helper function that is called after `expect(foo).not.toBeNull()` which in jasmine doesn't abort the rest of the expectations, so if we don't check for nulls here, the tests blows up if that expectation fails.

**Pull request closed without merging**